src/registriesView.ts

```diff
@@ -97,6 +104,50 @@ export class ComponentTypesView implements TreeDataProvider<ComponentType> {
         return this.registries;
     }
 
+    public getCompDescriptions(): Set<ComponentTypeDescription> {
+        return this.compDescriptions;
+    }
+
+    public getListOfRegistries(): Registry[] {
+        return this.registries;
+    }
+
+    public getAllComponents(): void {
+        let isError = false;
+        this.compDescriptions.clear();
+        void getInstance().getCompTypesJson().then(async (devFileComponentTypes: DevfileComponentType[]) => {
+            const components = new Set<string>(devFileComponentTypes.map((devFileComponentType: DevfileComponentType) => devFileComponentType.Name));
+            await this.getRegistries();
+            components.forEach((compName: string) => {
+                getInstance().execute(Command.describeCatalogComponent(compName)).then((componentDesc: CliExitData) => {
+                    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                    const out: ComponentTypeDescription[] = JSON.parse(componentDesc.stdout);
+                    out.forEach((component) => {
+                        // eslint-disable-next-line max-nested-callbacks
+                        component.Devfile?.starterProjects?.map((starter: StarterProject) => {
+                            starter.typeName = compName;
+                        });
+                        this.compDescriptions.add(component);
+                    });
+                    if (devFileComponentTypes.length === this.compDescriptions.size) {
+                        this.subject.next('refresh');
+                    }
+                }).catch(() => {
+                    isError = true;
+                }).finally(() => {
+                    if (isError) {
+                        if (this.subject.closed) {
+                            this.subject = new Subject<string>();
+                            this.getAllComponents();
```

Comment:
There is a risk of infinite cycle here. What if error persists? Ether exclude component from the list or mark it accordingly in UI.